components/dialog/dialog-mixin.js

```diff
@@ -315,10 +316,19 @@ export const DialogMixin = superclass => class extends RtlMixin(superclass) {
 		// native dialog backdrop does not prevent body scrolling
 		this._bodyScrollKey = preventBodyScroll();
 
+		// focus first focusable child prior to auto resize (fixes screen reader hiccups)
+		this._focusInitial();
+
 		requestAnimationFrame(async() => {
 			await this._updateSize();
 			this._state = 'showing';
-			this._focusInitial();
+
+			// edge case: no children were selected, try again after one redraw
+			if (!dialog.contains(document.activeElement)
```

Comment:
Does this check assume that there would be no other slots defined by the dialog?

Note, I think the composed active element is what this logic would be interested in, because if the dialog is wrapped in another web component and focus is within it, the `document.activeElement` would be that component, in the context of the outer document.

A safer approach that I think avoids these assumptions is to use the `getComposedActiveElement` and `isComposedAncestor` helpers - since it would be operating on the actual focused element, and it wouldn't matter whether it's within a slot or not.

Reply:
> ...if the dialog is wrapped in another web component and focus is within it, the document.activeElement would be that component

Good catch, this is the behavior I'm currently seeing. I've switched them out in favor of the existing helpers ([0a571485](https://github.com/BrightspaceUI/core/pull/1330/commits/0a57148588f91caadb556252cd234158b3cfb6de)).